### PR TITLE
Upgrade internal projects

### PR DESCRIPTION
**Pull request by amoerie**
Update: this is on hold until all contributors have had the chance to upgrade  their workstations to VS 2022

Because we still use .NET Core 2.1, and that has been out of support for a while now, I decided to slightly upgrade our internal dependencies. For .NET Core, I think we should target whatever is currently in LTS by Microsoft. Today, that is .NET Core 3.1 and .NET 6.0.
Of course, we still completely support .NET Framework, I'm talking about our test project and so on.

Note to @gofal and @mrbean-bremen: this does mean you will need to have Visual Studio 2022 installed to build the solution (or JetBrains Rider + .NET SDK 6). Proceed with caution. 
If, for some reason, you guys are unable to update Visual Studio yet, then we can postpone this PR.

Important: public projects that are packaged in NuGet were not modified

#### Checklist
- [x] The pull request branch is in sync with latest commit on the *fo-dicom/development* branch
- [x] I have updated API documentation
- [x] I have included unit tests
- [x] I have updated the change log
- [x] I am listed in the CONTRIBUTORS file

#### Changes proposed in this pull request:
- Upgrades FO-DICOM.Tests to .NET 4.6.2 + .NET Core 3.1 + .NET 6 
- Upgrades FO-DICOM.Dump to .NET 6.0
- Upgrades FO-DICOM.Benchmarks to .NET 6.0
- Upgrades FO-DICOM.Documentation.DocFx to .NET 6.0
- Upgrades codecs (from 5.0.0-beta6) to 5.2.0
- Upgrades infrastructural nuget dependencies (xUnit, ...)

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -9,8 +9,8 @@ on:
     - development
 
 jobs:
-  v5-net-framework:
-    runs-on: windows-2019
+  v5-net-462:
+    runs-on: windows-2022
```
Comment (mrbean-bremen) [inline .github/workflows/build.yml:13]: Maybe use windows-latest instead?

File: Tests/FO-DICOM.Tests/FO-DICOM.Tests.csproj [hunk 2]
```diff
@@ -64,15 +65,12 @@
   </ItemGroup>
 
   <ItemGroup>
-    <None Update="Test Data\10200904.dcm">
+    <None Include="Test Data\**\*.dcm">
```
Comment (mrbean-bremen) [inline Tests/FO-DICOM.Tests/FO-DICOM.Tests.csproj:77]: Good idea!

**Review by mrbean-bremen — APPROVED** [review]
Looks good to me! I just installed VS 2022 (though my notebook seemed not to like it, it crashed afterwards... but it's working anyway), so it's ok with me.

File: .github/workflows/build.yml (hunk 1 above)
Comment (amoerie) [reply .github/workflows/build.yml:13]: I had to upgrade to 2022 to get .NET 6 support, but I'm not sure I like the idea of our continuous integration being able to suddenly migrate to an entirely new version. 
Any particular reason why you're suggesting this?

Comment (mrbean-bremen) [reply .github/workflows/build.yml:13]: I used to use this to be able to see any problems with new versions, but you are right - it's better to have a controlled environment here.

**Comment by gofal** [comment]
VisualStudio 2022 was released in november. It takes some time to switch on my machine from studio 2019 to studio 2022. netcore 6 is already out for some time, but the lack of support in vusual studio 2019 was a blocker to try it in our company. 
Please give me some time until tos PR is merged!

**Comment by amoerie** [comment]
Take all the time you need @gofal there is no hurry.

I'll try to remember to ping you again in a few months then.

**Comment by mrbean-bremen** [comment]
@gofal - GH actions have now removed support for 2.1 (EOL since last August), so it makes sense to merge this soon, as all builds will fail now. If this is too soon, we could at least merge the subset that deals with 2.1.

**Comment by pengchen0692** [comment]
@amoerie Hello Alexander, wondering if there is any update for this PR? My PR https://github.com/fo-dicom/fo-dicom/pull/1354  is currently pending on this fix. :)

**Comment by amoerie** [comment]
It depends on whether @gofal  can compile .net 6 already or not. If not, then we create a separate PR that only removes .NET 2.1, if yes then we can simply merge this PR. 🙂

**Comment by gofal** [comment]
Question: in visual studio 2022 now my test-explorer now fail for the tests that require the transcoder in  net6.0 and net3.1 builds. in net462 it works fine.
they say: "System.DllNotFoundException : Unable to load DLL 'Dicom.Native-win64.dll' or one of its dependencies: Das angegebene Modul wurde nicht gefunden. (0x8007007E)"
bit the dll in located in bin\runtimes\win-x64\native. it was always located there and it works in studio 2019 and in net462.

any tip for me? how does it work on your workstations?

**Comment by mrbean-bremen** [comment]
Which tests are these? I'm running the tests right now on my notebook, with VS 2022 (Community edition), and ~so far~ all tests for net462, net6.0 and net3.1 have passed, except for the few intentionally skipped ones. I have done nothing special here.

**Comment by gofal** [comment]
it are 7 tests:
OutputLUTTest.ColorMap_Monochrome2ImageOptions_ReturnsMonochrome2ColorMap
DicomCodecExtensionsTest.ChangeTransferSyntax_FileFromRLELosslessToJPEGProcess2_4_WithParameters
DicomCodecExtensionsTest.ChangeTransferSyntax_FileFromRLELosslessToJPEGProcess2_4
DicomCodecExtensionsTest.ChangeTransferSyntax_FileFromJ2KToJPEGWithParameters_DoesNotThrow
DicomCodecExtensionsTest.ChangeTransferSyntax_DatasetFromJ2KToJPEGWithParameters_DoesNotThrow
GrayscaleRenderOptionsTest.ColorMap_Setter_ReturnsSetColorMap
GrayscaleRenderOptionsTest.ColorMap_Monochrome2ImageOptions_ReturnsMonochrome2ColorMap

All of them fail with the same exception as postet above: Unable to load the DLL Dicom.Native-win64.dll

:-(

I already did all the usual stuff (restarting visual studio, restarting workstation, clean all bin and obj directories and do a complete new build, ..) but nothing changed.

**Comment by mrbean-bremen** [comment]
Actually, I don't have any `Dicom.Native-win64.dll`. Wasn't that a library that fo-dicom used before the coding has been done by Efferent? Maybe you have some old stuff lying around that is still used unintentionally?

**Comment by mrbean-bremen** [comment]
Also note that these are all skipped for net462, so no wonder these did not fail. Do you have the correct version of the codecs (5.10) installed? Maybe you need a forced nuget restore?

**Comment by mrbean-bremen** [comment]
Actually, the DLL belongs to `fo-dicom.Codecs`, I hadn't looked correctly. I also have it in `runtimes\win-x64\native`, both under `net6.0` and under `netcoreapp3.1`.

**Comment by amoerie** [comment]
@gofal sorry to hear you're having trouble. 

- Is it possible you're missing a C++ redistributable to load these codecs?
- Are you running the tests in 32 bit or 64 bit?

I think the source code / dependencies should be fine because 
- it works in Github Actions
- it works on my machine in JetBrains Rider
- it works on my machine in VS

image

Can you verify if the following works on your machine:

```
dotnet test .\Tests\FO-DICOM.Tests\FO-DICOM.Tests.csproj --configuration Release --framework net6.0 --runtime win-x64
```

**Comment by mrbean-bremen** [comment]
It also works in Debug for me (both in VS and Rider), and I'm sure it has to do with the `fo-dicom.Codecs` package. Make sure to remove the nuget caches.

**Comment by gofal** [comment]
yes, i am also sure, that it must be some issue with my local installation of studio 2022.

**Comment by gofal** [comment]
finally managed to get green tests. Turned out, even though in VisualStudio I set the solution setting to x64, the projects where built in anyCPU and with AnyCPU the native codecs cannot be loaded. I had to add 

        <Platforms>AnyCPU;x64</Platforms>

to all the projects, so that visual studio actually builds then as x64. I wonder why it worked on your side.

**Comment by amoerie** [comment]
Can you try this?  Mine was set to "auto", maybe for some reason that doesn't work on your PC.

image

**Comment by gofal** [comment]
ah, mine is set to "x86". will change it to x64. 
Thats an very important information when setting up a new environment and building/testing fo-dicom.

**Comment by mrbean-bremen** [comment]
That's good to know! I had installed VS 2022 only recently, and didn't change anything, so this is a bit strange. Anyway, I was completely on the wrong track, kudos to @amoerie for figuring it out!

**Pull request merged**